packages/@angular/cli/tasks/init.ts

```diff
@@ -64,46 +58,54 @@ export default Task.extend({
       return Promise.reject(new SilentError(message));
     }
 
-    const blueprintOpts = {
-      dryRun: commandOptions.dryRun,
-      blueprint: 'ng',
-      rawName: packageName,
-      targetFiles: rawArgs || '',
-      rawArgs: rawArgs.toString(),
-      sourceDir: commandOptions.sourceDir,
-      style: commandOptions.style,
-      prefix: commandOptions.prefix.trim() || 'app',
-      routing: commandOptions.routing,
-      inlineStyle: commandOptions.inlineStyle,
-      inlineTemplate: commandOptions.inlineTemplate,
-      minimal: commandOptions.minimal,
-      ignoredUpdateFiles: ['favicon.ico'],
-      skipGit: commandOptions.skipGit,
-      skipTests: commandOptions.skipTests
-    };
-
     validateProjectName(packageName);
 
-    blueprintOpts.blueprint = normalizeBlueprint(blueprintOpts.blueprint);
+    const SchematicRunTask = require('../tasks/schematic-run').default;
+    const schematicRunTask = new SchematicRunTask({
+      ui: this.ui,
+      project: this.project
+    });
+
+    const cwd = this.project.root;
+    const schematicName = CliConfig.fromGlobal().get('defaults.schematics.newApp');
+
+    const runOptions = {
+      taskOptions: commandOptions,
+      workingDir: cwd,
+      collectionName: commandOptions.collectionName,
+      schematicName
+    };
 
-    return installBlueprint.run(blueprintOpts)
+    return schematicRunTask.run(runOptions)
+      .then(function () {
+        if (!commandOptions.dryRun) {
+          process.chdir(commandOptions.directory);
+        }
+      })
       .then(function () {
         if (!commandOptions.skipInstall) {
```

Comment:
It looks like a lapse that install still happens on a dry run. If only `schematicRunTask.run` is meant to happen on dry runs, it sounds better to wholly skip the promise chain after that when dry run is on.